frontend/sw360-portlet/src/main/resources/META-INF/resources/html/projects/includes/projects/clearingStatus.jspf

```diff
@@ -7,97 +7,315 @@
+        <div id="clearingStatusSpinner">
+            <%@ include file="/html/utils/includes/pageSpinner.jspf" %>
+        </div>
+        <table id="clearingStatusTable" class="table table-bordered d-none"></table>
     </div>
-</core_rt:if>
-<core_rt:if test="${releasesAndProjects.size()  > 0 }">
-    <table id="releasesTable" class="table table-bordered" data-details-context="${inProjectDetailsContext}"></table>
-</core_rt:if>
-
-<script type="text/javascript">
-require(['jquery', 'bridges/datatables', 'utils/render'], function ($, datatables, render) {
-    var homeUrl = themeDisplay.getURLHome().replace(/\/web\//, '/group/');
-
-    createClearingTable();
-
-    $('#exportReleasesSpreadsheetClearingButton').on('click', exportReleasesSpreadsheet);
-
-    function createClearingTable() {
-        var table,
-            result = [],
-            editRelease;
-
-        <core_rt:forEach items="${releasesAndProjects}" var="releasesClearingStatusData">
-            result.push({
-                "DT_RowId": "${releasesClearingStatusData.release.id}",
-                "0": "<sw360:DisplayReleaseLink release="${releasesClearingStatusData.release}" showFullname="true"/>",
-                "1": "<sw360:out value="${releasesClearingStatusData.projectNames}"/>",
-                "2": "<span id='releaseClearingState${releasesClearingStatusData.release.id}'><sw360:DisplayEnum value="${releasesClearingStatusData.release.clearingState}"/></span>",
-                "3": "<sw360:DisplayEnum value="${releasesClearingStatusData.release.mainlineState}"/>",
-                "4": "<sw360:out value="${releasesClearingStatusData.mainlineStates}"/>",
-                "5": "<sw360:DisplayEnum value="${releasesClearingStatusData.componentType}"/>",
-                "6": "${releasesClearingStatusData.release.id}"
-            });
-        </core_rt:forEach>
-
-        table = datatables.create('#releasesTable', {
-            data: result,
-            columns: [
-                {title: "<liferay-ui:message key="name" />"},
-                {title: "<liferay-ui:message key="project.origin" />"},
-                {title: "<liferay-ui:message key="clearing.state" />"},
-                {title: "<liferay-ui:message key="release.mainline.state" />"},
-                {title: "<liferay-ui:message key="project.mainline.state" />"},
-                {title: "<liferay-ui:message key="type" />"},
-                {title: "<liferay-ui:message key="actions" />", render: {display: renderActions}, className: "two actions" }
-            ],
-            columnDefs: [
-                {
-                    targets: [ 6 ],
-                    visible: $('#releasesTable').data() ? $('#releasesTable').data().detailsContext : false
-                }
-            ],
+</div>
+<%--for javascript library loading --%>
+<%@ include file="/html/utils/includes/requirejs.jspf" %>
+<script>
+AUI().use('liferay-portlet-url', function () {
```

Comment:
This javascript here looks huge. May be we can consider moving it into a separate js file.

Reply:
Tried moving script tag content to separate JS file.
Not working as expected , since it is using JSP features to resolve language tags and URLs

- The language properties are not getting resolved.
![screenshot1JPG](https://user-images.githubusercontent.com/56534710/81772209-e30d6300-9502-11ea-81ab-a758a1e399be.JPG)

- Also the Urls are not getting resolved for new as well as existing code
```
function downloadSourceCodeBundleButton(type) {
            var portletURL = Liferay.PortletURL.createURL('<%= PortletURLFactoryUtil.create(request, portletDisplay.getId(), themeDisplay.getPlid(), PortletRequest.RENDER_PHASE) %>');
            portletURL.setParameter('<%=PortalConstants.PROJECT_ID%>', '${project.id}');
            portletURL.setParameter('<%=PortalConstants.PAGENAME%>', '<%=PortalConstants.PAGENAME_SOURCE_CODE_BUNDLE%>');
            portletURL.setParameter('<%=PortalConstants.PROJECT_WITH_SUBPROJECT%>', type === 'projectWithSubProject' ? 'true' : 'false');

            window.location.href = portletURL.toString();
        }
```
![screenshot2](https://user-images.githubusercontent.com/56534710/81772202-ddb01880-9502-11ea-87f3-c244c88d5571.JPG)

Kindley suggest, if this can be achieved by any other means.